nixos/modules/services/video/rtsp-simple-server.nix

```diff
@@ -0,0 +1,70 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.rtsp-simple-server;
+  package = pkgs.rtsp-simple-server;
+  format = pkgs.formats.yaml {};
+in
+{
+  options = {
+    services.rtsp-simple-server = {
+      enable = mkEnableOption "RTSP Simple Server";
+
+      settings = mkOption {
+        description = ''
+          Settings for rtsp-simple-server.
+          Read more at <xlink:href="https://github.com/aler9/rtsp-simple-server/blob/main/rtsp-simple-server.yml"/>
+        '';
+        type = format.type;
+
+        default = {
+          logLevel = "info";
+          logDestinations = [
+            "stdout"
+          ];
+          logFile = "/var/log/rtsp-simple-server/rtsp-simple-server.log";
+        };
+      };
+
+      env = mkOption {
+        type = with types; attrsOf anything;
+        description = "Extra environment variables for RTSP Simple Server";
+        default = {};
+        example = {
+          RTSP_CONFKEY = "mykey";
+        };
+      };
+    };
+  };
+
+  config = mkIf (cfg.enable) {
+    environment.etc."rtsp-simple-server.yaml".source = format.generate "rtsp-simple-server.yaml" cfg.settings;
+
+    systemd.services.rtsp-simple-server = {
+      environment = cfg.env;
+      reloadIfChanged = true;
```

Comment:
not sure if this is right, prob not

application has abbility to reload when config changed, so config is now in /etc, but we'd still need to tell app it changed

Reply:
If you don't place the file under `/etc` then this will just magically work. By referencing the path in the nix store in `ExecStart` that gives your `.service` file a string that changes whenever the configuration file changes.

Please note that `reloadIfChanged` is usually a "bad idea" from a security standpoint. When a new version is released because of a major security vulnerability users don't get the update unless they reboot/manually restart the service.

Reply:
> By referencing the path in the nix store in ExecStart that gives your .service file a string that changes whenever the configuration file changes.

yes but wouldn't that restart the app instead of reloading it?

also need to check again if app might actually be smart enough to figure out the symlink changed

Reply:
`reloadIfChanged` prevents a service from restarting when your `.service` file changes and instead causes the service to reload.